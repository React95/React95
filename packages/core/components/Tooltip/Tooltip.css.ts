import { style } from '@vanilla-extract/css';
import { contract } from '../themes/contract.css';
import { calc } from '@vanilla-extract/css-utils';

export const tip = style({
  background: `radial-gradient(#ff0 20%, transparent 20%) 0 0,
     radial-gradient(#ff0 20%, transparent 20%) 4px 4px,
     radial-gradient(rgba(255, 255, 255, 0.1) 20%, transparent 25%) 0 1px,
     radial-gradient(rgba(255, 255, 255, 0.1) 20%, transparent 25%) 3px 4px`,
  backgroundSize: `${contract.space[7]} ${contract.space[7]}`,
  backgroundColor: contract.colors.borderLightest,
  borderWidth: contract.space[1],
  borderStyle: 'solid',
  paddingTop: contract.space[2],
  paddingRight: contract.space[2],
  paddingBottom: contract.space[2],
  paddingLeft: contract.space[4],
  boxShadow: 'none',
  position: 'absolute',
  top: calc.negate(contract.space[20]),
  textAlign: 'center',
  zIndex: contract.zIndices.taskbar,
});

export const wrapper = style({
  display: 'inline-block',
  position: 'relative',
  cursor: 'default',
  whiteSpace: 'nowrap',
});
