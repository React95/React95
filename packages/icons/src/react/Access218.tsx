import React from 'react';
import { IconProps } from '../iconType';
import { BaseIcon } from '../internal';
import Access218_32x32_4 from '../../png/Access218_32x32_4.png';

export const access218Data = {
  '32x32_4': {
    imageSrc: Access218_32x32_4 as string,
    width: 32,
    height: 32,
  },
};

export interface Access218Props extends IconProps {
  /**
   * Icon variant to use.
   * also provides default styling with the correct height and width
   **/
  variant?: '32x32_4';
}

export const Access218: React.FC<Access218Props> = ({
  variant = '32x32_4',
  ...rest
}) => {
  const image = access218Data[variant];

  return (
    <BaseIcon
      width={image.width}
      height={image.height}
      src={image.imageSrc}
      {...rest}
    />
  );
};
