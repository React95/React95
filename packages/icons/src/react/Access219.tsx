import React from 'react';
import { IconProps } from '../iconType';
import { BaseIcon } from '../internal';
import Access219_32x32_4 from '../../png/Access219_32x32_4.png';

export const access219Data = {
  '32x32_4': {
    imageSrc: Access219_32x32_4 as string,
    width: 32,
    height: 32,
  },
};

export interface Access219Props extends IconProps {
  /**
   * Icon variant to use.
   * also provides default styling with the correct height and width
   **/
  variant?: '32x32_4';
}

export const Access219: React.FC<Access219Props> = ({
  variant = '32x32_4',
  ...rest
}) => {
  const image = access219Data[variant];

  return (
    <BaseIcon
      width={image.width}
      height={image.height}
      src={image.imageSrc}
      {...rest}
    />
  );
};
