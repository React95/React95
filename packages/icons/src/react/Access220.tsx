import React from 'react';
import { IconProps } from '../iconType';
import { BaseIcon } from '../internal';
import Access220_32x32_4 from '../../png/Access220_32x32_4.png';

export const access220Data = {
  '32x32_4': {
    imageSrc: Access220_32x32_4 as string,
    width: 32,
    height: 32,
  },
};

export interface Access220Props extends IconProps {
  /**
   * Icon variant to use.
   * also provides default styling with the correct height and width
   **/
  variant?: '32x32_4';
}

export const Access220: React.FC<Access220Props> = ({
  variant = '32x32_4',
  ...rest
}) => {
  const image = access220Data[variant];

  return (
    <BaseIcon
      width={image.width}
      height={image.height}
      src={image.imageSrc}
      {...rest}
    />
  );
};
