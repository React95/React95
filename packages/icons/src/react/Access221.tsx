import React from 'react';
import { IconProps } from '../iconType';
import { BaseIcon } from '../internal';
import Access221_32x32_4 from '../../png/Access221_32x32_4.png';

export const access221Data = {
  '32x32_4': {
    imageSrc: Access221_32x32_4 as string,
    width: 32,
    height: 32,
  },
};

export interface Access221Props extends IconProps {
  /**
   * Icon variant to use.
   * also provides default styling with the correct height and width
   **/
  variant?: '32x32_4';
}

export const Access221: React.FC<Access221Props> = ({
  variant = '32x32_4',
  ...rest
}) => {
  const image = access221Data[variant];

  return (
    <BaseIcon
      width={image.width}
      height={image.height}
      src={image.imageSrc}
      {...rest}
    />
  );
};
