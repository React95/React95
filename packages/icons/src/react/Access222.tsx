import React from 'react';
import { IconProps } from '../iconType';
import { BaseIcon } from '../internal';
import Access222_32x32_4 from '../../png/Access222_32x32_4.png';

export const access222Data = {
  '32x32_4': {
    imageSrc: Access222_32x32_4 as string,
    width: 32,
    height: 32,
  },
};

export interface Access222Props extends IconProps {
  /**
   * Icon variant to use.
   * also provides default styling with the correct height and width
   **/
  variant?: '32x32_4';
}

export const Access222: React.FC<Access222Props> = ({
  variant = '32x32_4',
  ...rest
}) => {
  const image = access222Data[variant];

  return (
    <BaseIcon
      width={image.width}
      height={image.height}
      src={image.imageSrc}
      {...rest}
    />
  );
};
