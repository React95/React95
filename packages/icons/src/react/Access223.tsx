import React from 'react';
import { IconProps } from '../iconType';
import { BaseIcon } from '../internal';
import Access223_32x32_4 from '../../png/Access223_32x32_4.png';

export const access223Data = {
  '32x32_4': {
    imageSrc: Access223_32x32_4 as string,
    width: 32,
    height: 32,
  },
};

export interface Access223Props extends IconProps {
  /**
   * Icon variant to use.
   * also provides default styling with the correct height and width
   **/
  variant?: '32x32_4';
}

export const Access223: React.FC<Access223Props> = ({
  variant = '32x32_4',
  ...rest
}) => {
  const image = access223Data[variant];

  return (
    <BaseIcon
      width={image.width}
      height={image.height}
      src={image.imageSrc}
      {...rest}
    />
  );
};
