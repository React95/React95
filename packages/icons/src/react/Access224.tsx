import React from 'react';
import { IconProps } from '../iconType';
import { BaseIcon } from '../internal';
import Access224_32x32_4 from '../../png/Access224_32x32_4.png';

export const access224Data = {
  '32x32_4': {
    imageSrc: Access224_32x32_4 as string,
    width: 32,
    height: 32,
  },
};

export interface Access224Props extends IconProps {
  /**
   * Icon variant to use.
   * also provides default styling with the correct height and width
   **/
  variant?: '32x32_4';
}

export const Access224: React.FC<Access224Props> = ({
  variant = '32x32_4',
  ...rest
}) => {
  const image = access224Data[variant];

  return (
    <BaseIcon
      width={image.width}
      height={image.height}
      src={image.imageSrc}
      {...rest}
    />
  );
};
