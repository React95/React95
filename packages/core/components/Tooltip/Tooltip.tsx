import * as React from 'react';
import { tip, wrapper } from './Tooltip.css';
import cn from 'classnames';

function formatDate(date: Date): string {
  const monthNames = [
    'January',
    'February',
    'March',
    'April',
    'May',
    'June',
    'July',
    'August',
    'September',
    'October',
    'November',
    'December',
  ];

  const day = date.getDate();
  const monthIndex = date.getMonth();
  const year = date.getFullYear();

  return `${day.toString().padStart(2, '0')} ${monthNames[monthIndex]} ${year}`;
}

export type TooltipProps = {
  text?: string;
  delay?: number;
} & React.HTMLAttributes<HTMLDivElement>;

const TooltipRenderer = (
  {
    children,
    text = formatDate(new Date()),
    delay = 1000,
    ...rest
  }: TooltipProps,
  ref: React.Ref<HTMLDivElement>,
) => {
  const [show, setShow] = React.useState(false);
  const [delayTimer, setDelayTimer] = React.useState(0);

  const handleEnter = () => {
    const timer = window.setTimeout(() => {
      setShow(true);
    }, delay);

    setDelayTimer(timer);
  };

  const handleLeave = () => {
    clearTimeout(delayTimer);
    setShow(false);
  };

  return (
    <div
      {...rest}
      className={cn(wrapper, rest.className)}
      onMouseEnter={handleEnter}
      onMouseLeave={handleLeave}
      ref={ref}
    >
      {show && <div className={tip}>{text}</div>}
      {children}
    </div>
  );
};

export const Tooltip = React.forwardRef<HTMLDivElement, TooltipProps>(
  TooltipRenderer,
);
