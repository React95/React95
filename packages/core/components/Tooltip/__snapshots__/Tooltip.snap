// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Tooltip /> > Snapshot > should match snapshot 1`] = `
<div>
  <div
    class="Tooltip_wrapper__f5whxn1"
  >
    <span>
      hover
    </span>
  </div>
</div>
`;

exports[`<Tooltip /> > Snapshot > should match snapshot 2`] = `
<div>
  <div
    class="Tooltip_wrapper__f5whxn1"
  >
    <div
      class="Tooltip_tip__f5whxn0"
    >
      Tooltip text
    </div>
    <span>
      hover
    </span>
  </div>
</div>
`;
